server/processors-alerts-publisher/src/main/java/org/hawkular/apm/processor/alerts/Event.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright 2015-2016 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.apm.processor.alerts;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+import org.hawkular.apm.api.model.Property;
+import org.hawkular.apm.api.model.events.CompletionTime;
+
+/**
+ * @author Juraci Paixão Kröhling
+ */
+public class Event {
+    private String category;
+    private Map<String, String> context;
+    private Map<String, String> tags;
+    private long ctime;
+    private String id;
+    private String dataId;
+    private String dataSource;
+    private String text;
+
+    public Event() {
+    }
+
+    public Event(CompletionTime completionTime, String eventSource) {
+        if (null == eventSource || eventSource.isEmpty()) {
+            // TODO: should we instead just degrade the info we have? Like, by putting a generic value?
+            throw new IllegalStateException("Cannot determine the event source");
+        }
+        this.context = new HashMap<>(1);
+        this.context.put("id", completionTime.getId());
+
+        this.tags = new HashMap<>();
+        this.tags.put("uri", completionTime.getUri());
+        this.tags.put("operation", completionTime.getOperation());
+
+        Set<Property> properties = completionTime.getProperties();
+        if (properties != null && properties.size() > 0) {
+            properties.forEach(p -> this.tags.put(p.getName(), p.getValue()));
```

Comment:
just a note, but properties is a set because the property name may not be unique - so we need to consider how this may be handled when translating to a map for the alert tags. For now this should be ok.